store/store.go

```diff
@@ -531,7 +532,16 @@ func (s *Store) newRequest(reqOptions *requestOptions, user *auth.UserState) (*h
 		if err != nil {
 			return nil, err
 		}
-		// TODO: if device session is empty, request new session
+		if device.SessionMacaroon == "" {
+			err = s.refreshDeviceSession()
+			if err == state.ErrNoState {
+				// missing serial assertion, log and continue without device authentication
```

Comment:
ErrNoState shouldn't be here. Who's returning it? With such an error all we can tell is that something that was looking at *State did not find what it was expecting. It might be a terrible bug, or a completely different unexpected result that is not what the comment above says. We should have first class errors for those cases.